apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { isConfirmedTx } from '@alephium/shared'
+import { useInterval } from '@alephium/shared-react'
+import { useQuery } from '@tanstack/react-query'
+import { t } from 'i18next'
+import { X } from 'lucide-react'
+import { memo, useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+
+import { fadeInBottom, fadeOut } from '@/animations'
+import { pendingTransactionQuery } from '@/api/queries/transactionQueries'
+import Button from '@/components/Button'
+import HashEllipsed from '@/components/HashEllipsed'
+import { openModal } from '@/features/modals/modalActions'
+import { selectTopModal } from '@/features/modals/modalSelectors'
+import { sentTransactionStatusChanged } from '@/features/sentTransactions/sentTransactionsActions'
+import { selectSentTransactionByHash } from '@/features/sentTransactions/sentTransactionsSelectors'
+import SnackbarBox from '@/features/snackbar/SnackbarBox'
+import { useAppDispatch, useAppSelector } from '@/hooks/redux'
+import { SentTransaction } from '@/types/transactions'
+
+const SentTransactionSnackbarPopup = memo(({ txHash }: { txHash: string }) => {
+  const dispatch = useAppDispatch()
+  const sentTx = useAppSelector((s) => selectSentTransactionByHash(s, txHash))
+  const [hide, setHide] = useState(false)
+
+  const { data: tx } = useQuery(pendingTransactionQuery({ txHash, skip: !sentTx || sentTx.status === 'confirmed' }))
```

Comment:
Hm, I would go case by case. Not all our data fetching code is in the apiDataHooks. Some of it is in the features, like the chart for example.

Reply:
Ok. I just feel like fetching a pending TX is a pretty generic call (vs. feature specific). 
I still like the idea of this "fetching hook library" because it avoids duplicating the use of queries in component. What I mean by that: you develop a new feature, you need some data (say, a pending TX data), you look into this hook library (super easy to find what you're looking for thanks to our naming scheme) instead of relying on a custom `useQuery`.

But it's as you wish!

Reply:
I don't have a strong opinion, whether we do `useQuery(pendingTransactionQuery` or `useFetchPendingTx` is pretty much the same for me. We already have a library of queries where `pendingTransactionQuery` is defined. I try to avoid unnecessary boilerplate code if it's not needed.

We have 2 uses of this query currently:

```ts
const { data: tx } = useQuery(pendingTransactionQuery({ txHash, skip: !sentTx || sentTx.status === 'confirmed' }))
```

```ts
const { data: pendingTx } = useQuery(pendingTransactionQuery({ txHash, skip: !isPendingTx }))
```

If you want, I can create a `useFetchPendingTransaction` that accepts the same params as the `pendingTransactionQuery` and it's simply a wrapper around `useQuery`.

Reply:
I think I would like it! :)

Reply:
ok, so you'd like an entire new file just for this :P 

<img width="1014" alt="image" src="https://github.com/user-attachments/assets/7ccdb0a7-f2c6-400a-bf00-8b19cf13ada8">

Yes, my lord. It shall be done.

Reply:
I thought we were gonna use Tanstack to reduce Redux boilerplate code and now we are adding our own haha

Reply:
Done in f6cc222ed